test/functional/test-iframe-helper.js

```diff
@@ -134,20 +134,24 @@ describe('iframe-helper', function() {
     });
   });
 
-  it('should un-listen after first hit', () => {
+  it('should un-listen and resolve promise after first hit', () => {
     let calls = 0;
     return new Promise(resolve => {
-      IframeHelper.listenForOnce(testIframe, 'send-intersections', () => {
-        calls++;
-        resolve();
-      });
+      IframeHelper.listenForOncePromise(testIframe,
+          ['no-msg', 'send-intersections', 'send-intersections'])
```

Comment:
hmm, having the duplicated message is not a valid use case. the behavior is in-deterministic and we don't really care, so there's no need to have a test to protect it.